test/lib/config/index.js

```diff
@@ -0,0 +1,701 @@
+/*jshint sub:true*/
+'use strict';
+
+const path = require('path');
+const url = require('url');
+const _ = require('lodash');
+const proxyquire = require('proxyquire');
+
+const ConfigReader = require('../../../lib/config/config-reader');
+const defaults = require('../../../lib/config/defaults');
+const Config = require('../../../lib/config');
+
+describe('config', () => {
+    const sandbox = sinon.sandbox.create();
+
+    afterEach(() => sandbox.restore());
+
+    const mkConfig_ = (opts) => {
+        return _.defaults(opts || {}, {
+            specs: ['path/to/test']
+        });
+    };
+
+    describe('parse', () => {
+        let configStub;
+        let parseOptions;
+
+        beforeEach(() => {
+            parseOptions = sandbox.stub();
+
+            configStub = proxyquire('../../../lib/config', {
+                './options': parseOptions
+            });
+        });
+
+        it('should read config file', () => {
+            sandbox.stub(ConfigReader.prototype, 'read').returns({});
+
+            configStub.create({}).parse();
+
+            assert.isTrue(ConfigReader.prototype.read.calledOnce);
+        });
+
+        it('should set dir path for config file to projectRoot field', () => {
+            sandbox.stub(ConfigReader.prototype, 'read').returns({config: 'foo/bar/.config.js'});
+
+            configStub.create({}).parse();
+
+            assert.propertyVal(parseOptions.firstCall.args[0].options, 'projectRoot', 'foo/bar');
+        });
+
+        it('should set curr working dir to projectRoot field when config file does not exist', () => {
+            sandbox.stub(ConfigReader.prototype, 'read').returns({});
+
+            configStub.create({}).parse();
+
+            assert.propertyVal(parseOptions.firstCall.args[0].options, 'projectRoot', process.cwd());
+        });
+    });
+
+    describe('overrides options', () => {
+        beforeEach(() => sandbox.stub(ConfigReader.prototype, 'read'));
+
+        afterEach(() => {
+            delete process.env['hermione_base_url'];
+            process.argv = [];
+        });
+
+        it('should not override anything by default', () => {
+            const readConfig = mkConfig_({baseUrl: 'http://default.com'});
```

Comment:
Yes, i know. But from my point of view it makes each test case more understandable. I mean i don't need to scroll up and look what the `baseUrl` is used